src/main/java/com/dev/cinema/service/impl/UserServiceImpl.java

```diff
@@ -0,0 +1,26 @@
+package com.dev.cinema.service.impl;
+
+import com.dev.cinema.dao.UserDao;
+import com.dev.cinema.library.Inject;
+import com.dev.cinema.library.Service;
+import com.dev.cinema.model.User;
+import com.dev.cinema.service.UserService;
+import com.dev.cinema.util.HashUtil;
+
+@Service
+public class UserServiceImpl implements UserService {
+    @Inject
+    private UserDao userDao;
+
+    @Override
+    public User add(User user) {
+        user.setSalt(HashUtil.getSalt());
```

Comment:
You have already initialized this field in AuthenticationServiceImpl register method

Reply:
done